haddock-api/src/Haddock/Backends/Xhtml/Decl.hs

```diff
@@ -646,7 +646,7 @@ ppInstanceSigs links splice unicode qual sigs = do
     TypeSig lnames typ <- sigs
     let names = map unLoc lnames
         L loc rtyp = get_type typ
-    return $ ppSimpleSig links splice unicode qual loc names rtyp
+    return $ ppSimpleSig links splice unicode qual (getLoc $ head $ lnames) names rtyp
```

Comment:
@niteria can you explain this change here? Would it make sense to extend the `sinify` business to give more meaningful src locs if possible to avoid not doing the obvious thing here?